lib/svrquery/client.go

```diff
@@ -34,28 +39,28 @@ type Client struct {
 
 // WithKey sets the key used for request by for the client.
 func WithKey(key string) Option {
-	return func(c *Client) error {
+	return func(c *UDPClient) error {
 		c.key = key
 		return nil
 	}
 }
 
 // WithTimeout sets the read and write timeout for the client.
 func WithTimeout(t time.Duration) Option {
-	return func(c *Client) error {
+	return func(c *UDPClient) error {
 		c.timeout = t
 		return nil
 	}
 }
 
-// NewClient creates a new client that talks to addr.
-func NewClient(proto, addr string, options ...Option) (*Client, error) {
+// NewUDPClient creates a new client that talks to addr.
+func NewUDPClient(proto, addr string, options ...Option) (*UDPClient, error) {
```

Comment:
I think this is going to break users of the library as NewClient is publicly exported and the main way people will interact with the client.

Can we either have the struct do both or abstract it internally?